src/lib.rs

```diff
@@ -1658,6 +1658,24 @@ impl std::ops::DerefMut for BufferViewMut<'_> {
     }
 }
 
+impl AsRef<[u8]> for BufferView<'_> {
+    fn as_ref(&self) -> &[u8] {
+        self.data
+    }
+}
+
+impl AsRef<[u8]> for BufferViewMut<'_> {
```

Comment:
can we remove that one?

Reply:
Oh, good point. I didn't notice that the deref impl for the write view panics. But that makes sense.

Will remove :)

Reply:
Hmm, but doesn't Rust still allow read access through a mut reference? That still leaves a soundness hole, but that's also true for `DerefMut` then.

Reply:
I don't think reading a write-only buffer is unsafe. It may just be very slow.

Reply:
Rust allows that, but we don't want to pretend it's ok for this case. People can jump hoops to do it, still, but hopefully they will not :)

Reply:
Right, as long as it's not unsafe. Otherwise it might be a good idea to make the functions that create buffer views unsafe.

But I don't know enough about GPU memory mapping to know what makes sense, so I'll happily defer to your collective wisdom :)